nova_vm/src/ecmascript/builtins/promise/data.rs

```diff
@@ -8,23 +8,36 @@ use crate::{ecmascript::{builtins::control_abstraction_objects::promise_objects:
 pub struct PromiseHeapData {
     pub(crate) object_index: Option<OrdinaryObject>,
     pub(crate) promise_state: PromiseState,
```

Comment:
**praise:** It's very pretty seeing the whole promise feature set moving into the promise state. "All things as they should be" kind of thing.